packages/db/src/query/live/collection-subscriber.ts

```diff
@@ -80,18 +82,13 @@ export class CollectionSubscriber<
       this.collection.config.getKey
     )
 
-    // Do not provide the callback that loads more data
-    // if there's no more data to load
-    // otherwise we end up in an infinite loop trying to load more data
     const dataLoader = sentChanges > 0 ? callback : undefined
 
-    // We need to call `maybeRunGraph` even if there's no data to load
-    // because we need to mark the collection as ready if it's not already
-    // and that's only done in `maybeRunGraph`
```

Comment:
This comment is also useful to keep but may need to be changed slightly (at least the references to `maybeRunGraph` need to be changed to `scheduleGraphRun`)